api/client/api_client.go

```diff
@@ -0,0 +1,139 @@
+package client
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/palantir/stacktrace"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+const(
+	personEndpoint = "person"
+	textContentType = "text/plain"
+
+	timeoutSeconds = 2 * time.Second
+	incrementBooksReadEndpoint = "incrementBooksRead"
+
+	healthcheckUrlSlug = "health"
+	healthyValue       = "healthy"
+)
+
+type Person struct {
+	BooksRead int
+}
+
+type APIClient struct {
+	httpClient http.Client
+	ipAddr     string
+	port       int
+}
+
+func NewAPIClient(ipAddr string, port int) *APIClient {
+	httpClient := http.Client{
+		Timeout: timeoutSeconds,
+	}
+	return &APIClient{
+		httpClient: httpClient,
+		ipAddr:     ipAddr,
+		port:       port,
+	}
+}
+
+func (client *APIClient) getPersonUrlForId(id int) string {
```

Comment:
Note on this: I intentionally leave the receiver to be by-value when I don't expect a function to modify the internal state of the object. This provides extra defensiveness, so that people don't accidentally modify the internal state when they don't want to.

Reply:
Second convention: we organize Go code to put all public functions first, followed by all private functions (which helps organize the code a bit). You had no idea about that, of course, so I just updated https://github.com/kurtosis-tech/kurtosis-internal-docs/blob/master/go-style-guide.md to reflect it xD

Reply:
Yes, I had an idea about this, but in this case, I forgot to check the order.